### PR TITLE
chore: Update `EWS` package

### PR DESCRIPTION
**Pull request by joeauyeung**
## What does this PR do?

This PR fixes our Exchange integration by doing the following:
- Updating our EWS package from 1.4.4 -> 3.1.2
- Removes the Exchange version field if NTLM authentication is being used

https://www.loom.com/share/541e3871c78445929bd8fd66ea3925cd

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

### CODE REVIEW

File: packages/app-store/exchangecalendar/pages/setup/index.tsx [hunk 1]
```diff
@@ -119,7 +120,7 @@ export default function ExchangeSetup() {
                         <SelectField
                           label={t("exchange_authentication")}
                           options={authenticationMethods}
-                          defaultValue={authenticationMethods[0]}
+                          defaultValue={authenticationMethods[1]}
```
Comment (hbjORbj) [inline packages/app-store/exchangecalendar/pages/setup/index.tsx:123]: nit: can we store this in a variable with a self-explanatory name

File: packages/app-store/exchangecalendar/pages/setup/index.tsx [hunk 2]
```diff
@@ -129,23 +130,25 @@ export default function ExchangeSetup() {
...
+                    {authenticationMethod === ExchangeAuthentication.STANDARD ? (
+                      <Controller
+                        name="exchangeVersion"
+                        control={form.control}
+                        render={({ field: { onChange } }) => (
+                          <SelectField
+                            label={t("exchange_version")}
+                            options={exchangeVersions}
+                            defaultValue={exchangeVersions[7]}
+                            onChange={async (version) => {
```
Comment (hbjORbj) [inline packages/app-store/exchangecalendar/pages/setup/index.tsx:142]: async needed here?

File: packages/app-store/exchangecalendar/pages/setup/index.tsx [hunk 3]
```diff
@@ -129,23 +130,25 @@ export default function ExchangeSetup() {
...
-                    />
+                    {authenticationMethod === ExchangeAuthentication.STANDARD ? (
+                      <Controller
+                        name="exchangeVersion"
+                        control={form.control}
+                        render={({ field: { onChange } }) => (
+                          <SelectField
+                            label={t("exchange_version")}
+                            options={exchangeVersions}
+                            defaultValue={exchangeVersions[7]}
```
Comment (hbjORbj) [inline packages/app-store/exchangecalendar/pages/setup/index.tsx:141]: nit: can we store this in a variable with a self-explanatory name

**Review by hbjORbj — DISMISSED** [review]
Approved with code nit comments. Look into the type-check test btw

**Comment by joeauyeung** [comment]
Removed instances of `gzip` in https://github.com/calcom/cal.com/pull/16950/commits/28deab1d4fb525572a1b9c7fdc4c085edafd6299 since in the constructor of the `XhrApi` class it didn't even take into account this param.

https://github.com/ewsjs/xhr/blob/5a40d48be49bf0b7a57e9bca128d7cc4824f99b3/src/xhrApi.ts#L277-L228

File: packages/app-store/exchangecalendar/pages/setup/index.tsx (hunk 1 above)
Comment (emrysal) [reply packages/app-store/exchangecalendar/pages/setup/index.tsx:123]: Not really, but will update with a better version.

**Comment by hbjORbj** [comment]
Screenshot 2024-10-09 at 21 32 55

I installed this version of `deasync` directly as dev dependency to fix the build error. The new version of `http-cookie-agent` used by the new version of EWS package seems tor require this specific version

**Review by emrysal — APPROVED** [review]
LGTM!

**Pull request merged**